### ISSUE TITLE
Inline raw starting with dot needlessly prevents line break before itself

### ISSUE THREAD

**Issue by MDLC01** [typst/typst#5317]
### Description

Images generated by the Discord bot.

```typ
Test overflow
#pagebreak()
Test overflow `.dot`
```
![image](https://github.com/user-attachments/assets/c635df1c-19b3-4f7b-9726-b39d4818bcfb)
![image](https://github.com/user-attachments/assets/554a50c4-9ed4-4906-af18-65fef638221b)

### Typst version

- [X] I am using the latest version of Typst

**Comment by EpicEricEE** [comment]
This doesn't need the raw, it is also the case for normal text and isn't limited on dots, but on all kinds of "ending" punctuation (e.g. commas, question marks, exclamation marks, closing parentheses):

```typ
#set page(width: 3cm, height: auto)
Test overflow !dot
```

**Comment by laurmaedje** [comment]
That's just how the Unicode linebreaking algorithm works. I'm not sure deviating from it is advisable.

**Comment by MDLC01** [comment]
> This doesn't need the raw

Indeed, except in cases where a `raw` element is used, I think it is clear that allowing a line break isn't problematic.

**Comment by laurmaedje** [comment]
So what you desire is basically an unconditional linebreak opportunity around raw?

**Comment by MDLC01** [comment]
Maybe only if there are spaces around the inline raw, but essentially, yes.

This is similar to how it works for inline math right now I believe:
```typ
#set page(width: 3cm, height: auto)
Test overflow $.x$
```
![".x" is alone on its line.](https://github.com/user-attachments/assets/4420f308-38af-4a9d-813c-922fb514a8b1)

**Comment by MDLC01** [comment]
This issue is visible in the `raw` syntax list added by #8624, in the PDF documentation.

<img width="1240" height="896" alt="A screenshot of the PDF documentation where a line containing a lot of raw elements starting with periods overflows." src="https://github.com/user-attachments/assets/243f1459-9d2b-4796-a9f7-2821350c2fe9" />

**Comment by saecki** [comment]
A counter example where this is behavior is desirable would be when referencing the plural of a type. Such as the `ListItem`s of a `ListElem`.  Allowing a linebreak before the "s" outside of the raw text wouldn't be great.
Though I see this argument isn't especially strong.

**Comment by laurmaedje** [comment]
> Though I see this argument isn't especially strong.

Is it not?

**Comment by bastienvoirin** [comment]
> A counter example where this is behavior is desirable would be when referencing the plural of a type. Such as the `ListItem`s of a `ListElem`. Allowing a linebreak before the "s" outside of the raw text wouldn't be great.

IIUC what @MDLC01 said above we should only allow linebreaks if there are spaces before _and after_ the inline raw.

**Comment by MDLC01** [comment]
> A counter example where this is behavior is desirable would be when referencing the plural of a type.

A similar example that I find even more convincing is an inline `raw` followed by punctuation.
